cmd/juju/application/addrelation.go

```diff
@@ -24,34 +24,157 @@ import (
 )
 
 const addRelationDoc = `
-Add a relation between 2 local application endpoints or a local endpoint and a remote application endpoint.
-Adding a relation between two remote application endpoints is not supported.
-
-Application endpoints can be identified either by:
-    <application name>[:<relation name>]
-        where application name supplied without relation will be internally expanded to be well-formed
-or
-    <model name>.<application name>[:<relation name>]
-        where the application is hosted in another model owned by the current user, in the same controller
-or
-    <user name>/<model name>.<application name>[:<relation name>]
-        where user/model is another model in the same controller
-
-For a cross model relation, if the consuming side is behind a firewall and/or NAT is used for outbound traffic,
-it is possible to use the --via option to inform the offering side the source of traffic so that any required
-firewall ports may be opened.
+Relate two applications. Relations are a communication interface provided by 
+the Juju controller that enable units to transfer data. This star messaging 
```

Comment:
My first though was to break this out and elaborate, instead of having the term "star messaging topology" fly by.

But then, thinking about it more, should this even be presented to the user out of the gate? The reason I ask is that although Juju does indeed use this pathway for a relation, the modelling for the operator's benefit is to present it as a direct association - we're mixing the concept with the implementation.

Gustavo mentioned this exact thing - the model versus the implementation in meetings we had with the charm-tech team in Frankfurt.

Reply:
I have a lot of sympathy for this view. After several iterations going backwards and forwards, I settled on including it.

Some rationale. Let me know if you find it at all persuasive...

We need to demystify relations. Our language implies that there is a connection between application<->application. But it's unit<->unit. That mismatch creates problems when users need to start to write charms and/or understand peer relations, leadership.

So this shifts us towards describing things in terms of unit<->unit. At least briefly. But we now have another problem. Unit agents don't connect to one another. So how is this possible? At this point, it's important to mention the role of the controller.

Using "star typology" is jargon. But it's an established term that's easy to search for. Unlike, ahem, "relations".

We don't have a principled stance on when to elide implementation details. We expose the unit and machine agents' status by default, for example.

Reply:
@manadart did you have any further thoughts on this?

Reply:
I'm good with it. Let's get it landed while you're here.